pygmt/src/histogram.py

```diff
@@ -0,0 +1,63 @@
+"""
+Histogram - Create a histogram
+"""
+from pygmt.clib import Session
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    A="horizontal",
+    B="frame",
+    C="cmap",
+    G="fill",
+    J="projection",
+    R="region",
+    T="interval",
+    W="pen",
+    p="perspective",
+)
+@kwargs_to_strings(R="sequence", T="sequence")
+def histogram(self, table, **kwargs):
+    r"""
+    Plots a histogram, and can read data from a file or
+    list, array, or dataframe.
+
+    Full option list at :gmt-docs:`histogram.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+
+    table : str, list, or 1d array
+        A data file name, list, or 1d numpy array. This is a required argument.
+    {J}
+    {R}
+    {B}
+    {CPT}
+    {G}
+    {W}
+    {p}
+    horizontal : bool
+        Plot the histogram using horizonal bars instead of the
+        default vertical bars.
+    interval : int or str
+        [*min*\ /*max*\ /]\ *inc*\ [**+n**\ ]
+        Set the interval for the width of each bar in the histogram.
+
+    """
```

Comment:
Need to add `self.preprocess` to ensure histogram is plotted to the right figure always.

```suggestion
    """
    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
```